h3-iroh/examples/server.rs

```diff
@@ -74,15 +74,19 @@ async fn main() -> Result<()> {
 
 async fn handle_connection(incoming: Incoming, root: Arc<Option<PathBuf>>) -> Result<()> {
     let conn = incoming.accept()?.await?;
-    let remote_node_id = conn.remote_node_id()?;
+    let remote_endpoint_id = conn.remote_id()?;
     let span = Span::current();
-    span.record("remote_node_id", remote_node_id.fmt_short().to_string());
+    span.record(
+        "remote_endpoint_id",
+        remote_endpoint_id.fmt_short().to_string(),
+    );
     info!("new connection");
 
     let mut h3_conn = h3::server::Connection::new(h3_iroh::Connection::new(conn)).await?;
     loop {
         match h3_conn.accept().await {
-            Ok(Some((req, stream))) => {
+            Ok(Some(req_resolver)) => {
+                let (req, stream) = req_resolver.resolve_request().await?;
```

Comment:
I think you want to log this error and continue the loop, instead of bailing using `?`?  I think this replaces the `StreamError` case from before